frontend/public/components/storage/create-pvc.tsx

```diff
@@ -83,7 +86,9 @@ export class CreatePVCForm extends React.Component<CreatePVCFormProps, CreatePVC
 
   handleStorageClass = storageClass => {
     //if the provisioner is unknown or no storage class selected, user should be able to set any access mode
-    const modes = storageClass && storageClass.provisioner && provisionerAccessModeMapping[storageClass.provisioner] ? provisionerAccessModeMapping[storageClass.provisioner] : ['ReadWriteOnce', 'ReadWriteMany', 'ReadOnlyMany'];
+    const modes = storageClass && storageClass.provisioner && provisionerAccessModeMapping[storageClass.provisioner] ? provisionerAccessModeMapping[storageClass.provisioner] : cephStorageProvisioners.some(
+      (provisioner: string) => storageClass.provisioner.includes(provisioner),
+    ) ? ['ReadWriteOnce', 'ReadWriteMany'] : ['ReadWriteOnce', 'ReadWriteMany', 'ReadOnlyMany'];
```

Comment:
Don't nest ternary operators. It's extremely difficult to follow.

Reply:
ack